sklearn/datasets/value_dropper.py

```diff
@@ -0,0 +1,300 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+import numbers
+
+from sklearn.utils import check_array
+from sklearn.utils import check_random_state
+from sklearn.utils.multiclass import type_of_target
+
+from sklearn.base import TransformerMixin
+from sklearn.preprocessing import LabelEncoder
+
+
+__all__ = ["ValueDropper"]
+
+
+class ValueDropper(TransformerMixin):
+    """Artificially insert NMAR or MCAR missing values into data.
+
+    Where,
+
+    NMAR/MNAR - Not Missing At Random / Missing Not At Random
+        When the missingness is correlated with the class classes in the
+        target (y) (and hence informative).
+
+    MCAR - Missing Completely At Random
+        When the missingness is completely random (and hence uninformative).
+
+    If the missing type is NMAR, a ``missing_proba`` parameter can be passed
+    to drop values conforming to the given drop-probabilities.
+
+
+    Parameters
+    ----------
+
+    missing_values : {"NaN" (or np.nan) | int | float}, default "NaN"
+        The value to insert to indicate missingness.
+
+    missing_proba : dict of floats or dict of vector of floats
+        To vary the proportion of values dropped across each feature,
+        individual drop-probabilities for each feature can be specified as a 1D
+        array-like of shape (n_features, ) (e.g. [0.1, 0.15, 0.1]).
+
+        If missingness is not MCAR, a dict of floats can be used to specify
+        the drop-probabilities on a per-label basis
+        (e.g. {1: 0.2, 2: 0.3, 3: 0.5}).
+
+        This dict can also contains some 1D array-like of shape (n_features, )
+        to vary drop-probabilities across features
+        (e.g. {1: 0.1, 3: [0.1, 0.15, 0.1]}).
+
+    copy : bool, default False
+        Whether to copy the data or work inplace.
+
+    random_state : int, optional
+        The seed for the numpy's random number generator.
+
+        If ``random_state`` is set to an integer, the ``missing_proba``
+        can be upscaled safely with the assumption that all the values
+        dropped with a smaller scale will exist in the larger scaled version::
+            missing_proba_1 = {0: 0.1, 3: [0.3, 0.1, 0.1]}
+            missing_proba_2 = {0: 0.1, 1:0.2, 3: [0.6, 0.1, 0.8]}
+
+        The missing values dropped with ``missing_proba_1`` will also
+        be dropped with ``missing_proba_2``.
+
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> X = np.array([[0., 1., 2.],
```

Comment:
With < 5 samples I find it difficult to clearly show how we can vary distribution by classes and by features...